### ISSUE TITLE
Star operator without enclosing within parentheses

### ISSUE THREAD

**Issue by thorwhalen** [psf/black#2496]
**Describe the bug**

Returning a tuple containing star operator without enclosing with parentheses leads to a `Cannot parse` error.

Oh no! 💥 💔 💥

**To Reproduce**

```python
def apply_to_last(input, func):
    *header, last = input
    return *header, func(last)

# which is valid code doing this:
assert apply_to_last([1,2,3], lambda x: x+10) == (1, 2, 13)
```

And try to blackify...

**Expected behavior**

I expect it to not choke. 

There is a *workaround* -- enclosing in parentheses

```python
def apply_to_last(input, func):
    *header, last = input
    return (*header, func(last))
```

But doing this makes the linter complain that _"Remove redundant parentheses"._

**Environment (please complete the following information):**

- Version: `axblack 20210720`
- MacOS 10.15.7
- Python 3.8.6

**Comment by JelleZijlstra** [comment]
What python version do you have in your settings? This code is invalid in 3.6.

**Comment by thorwhalen** [comment]
As in `python --version` right? 

I ask because I mentioned it above:

```
3.8.6
```

**Comment by JelleZijlstra** [comment]
No, I mean Black's target-version setting. By default Black assumes your code is 3.6-compatible. Does it work if you limit your target-version to 3.8+.

We could perhaps autodetect this grammar change though and detect that the code is 3.8+.
